### PR TITLE
Extend quadratic_denom_rule in manual integration

### PR DESCRIPTION
**Pull request by miguelmarco**
#### Brief description of what is fixed or changed

Currently there are no rules that can handle integrals of the form (a*x+b)/(c*x**2+d*x+e).This PR adds the usual rewriting and change of variable used to integrate these cases:

```
>>> from sympy import *
>>> x = S('x')
>>> f = (5*x + 2)/(3*x**2 - 2*x + 8)
>>> manualintegrate.manualintegrate(f, x)
5*log(3*x**2 - 2*x + 8)/6 + 11*sqrt(23)*atan(3*sqrt(23)*(x - 1/3)/23)/69
>>> manualintegrate.integral_steps(f, x)
RewriteRule(rewritten=(5*x - 5/3)/(3*x**2 - 2*x + 8) + 11/(3*(3*x**2 - 2*x + 8)), substep=AddRule(substeps=[ConstantTimesRule(constant=5/6, other=(6*x - 2)/(3*x**2 - 2*x + 8), substep=URule(u_var=_u, u_func=3*x**2 - 2*x + 8, constant=5/6, substep=ReciprocalRule(func=_u, context=1/_u, symbol=_u), context=(5*x + 2)/(3*x**2 - 2*x + 8), symbol=x), context=(5*x - 5/3)/(3*x**2 - 2*x + 8), symbol=x), ConstantTimesRule(constant=11/3, other=1/(3*x**2 - 2*x + 8), substep=URule(u_var=_u, u_func=x - 1/3, constant=None, substep=AlternativeRule(alternatives=[RewriteRule(rewritten=3/(9*_u**2 + 23), substep=ConstantTimesRule(constant=3, other=1/(9*_u**2 + 23), substep=PiecewiseRule(subfunctions=[(ArctanRule(a=1, b=9, c=23, context=1/(9*_u**2 + 23), symbol=_u), True), (ArccothRule(a=1, b=9, c=23, context=1/(9*_u**2 + 23), symbol=_u), False), (ArctanhRule(a=1, b=9, c=23, context=1/(9*_u**2 + 23), symbol=_u), False)], context=1/(9*_u**2 + 23), symbol=_u), context=3/(9*_u**2 + 23), symbol=_u), context=1/(-2*_u + 3*(_u + 1/3)**2 + 22/3), symbol=_u), RewriteRule(rewritten=1/(3*_u**2 + 23/3), substep=PiecewiseRule(subfunctions=[(ArctanRule(a=1, b=3, c=23/3, context=1/(3*_u**2 + 23/3), symbol=_u), True), (ArccothRule(a=1, b=3, c=23/3, context=1/(3*_u**2 + 23/3), symbol=_u), False), (ArctanhRule(a=1, b=3, c=23/3, context=1/(3*_u**2 + 23/3), symbol=_u), False)], context=1/(3*_u**2 + 23/3), symbol=_u), context=1/(-2*_u + 3*(_u + 1/3)**2 + 22/3), symbol=_u)], context=1/(-2*_u + 3*(_u + 1/3)**2 + 22/3), symbol=_u), context=1/(-2*_u + 3*(_u + 1/3)**2 + 22/3), symbol=x), context=11/(3*(3*x**2 - 2*x + 8)), symbol=x)], context=(5*x + 2)/(3*x**2 - 2*x + 8), symbol=x), context=(5*x + 2)/(3*x**2 - 2*x + 8), symbol=x)
```

#### Release Notes

* integrals
  * added rules in manualintegrate to cover all cases of rational functions with quadratic denominator

### CODE REVIEW

File: sympy/integrals/manualintegrate.py [hunk 1]
```diff
@@ -733,19 +733,62 @@ def quadratic_denom_rule(integral):
...
+    if match2:
+        b, c =  match2[b], match2[c]
+        if b.is_zero:
+            return
+        u = sympy.Dummy('u')
+        u_func = symbol + c/(2*b)
+        integrand2 = integrand.subs(symbol, u - c / (2*b))
+        next_step = integral_steps(integrand2, u)
+        if next_step:
+            return URule(u, u_func, None, next_step, integrand2, symbol)
```
Comment (jksuom) [inline sympy/integrals/manualintegrate.py]: What will happen if there is no `next_step`. Now `b` and `c` are not wild symbols so `match3` may not work as expected.

Comment (miguelmarco) [reply sympy/integrals/manualintegrate.py]: I don't think that is possible, but just to make sure, I added a line to handle that case.

**Comment by oscargus** [comment]
I restarted the failing test (which was due to http-errors).

Tests are required for the change. The one you gave in the first post is a good start. (A hint is to write 1/3 as S(1)/3 to avoid having to rewrite it at a later commit...)

Btw, your provided example already works with `integrate`. I've never really understood if `manualintegrate`is useful on its own or if it is just a way to deal with specific types of integrals that cannot be handled elsewhere. (I've earlier tried to figure out which rules that can be removed, as much of the functionality probably have been introduced elsewhere.) Hence, I would ideally see an example of an integral that can now be handled by integrate that was not possible before (or some other motivation why manualintegrate is valuable as such, it may very well be, just that I do not know about it).

**Comment by miguelmarco** [comment]
I am on travel until the end of the week, will work on the tests then.

My understanding is that manualintegrate is mostly conceived with an educational purpose: it tries to follow the steps that a calculus student would follow (and in fact the steps can be shown) and hence produce an answer that looks like the one a student would obtaining. It is very used in the sympygamma site

**Comment by jksuom** [comment]
Thanks, this looks good.

**Pull request merged**